### PR TITLE
Added Metered Button

### PR DESCRIPTION
**Pull request by GoodHood1**

### CODE REVIEW

File: Manual/HTML/Metered Button.html [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+                    <tr><td>3</td><td>H3</td><td>M7</td><td>M6</td><td>H2</td><td>M3</td><td>T5</td><td>T2</td><td>X4</td><td>T2</td><td>H7</td></tr>
+                    <tr><td>4</td><td>T4</td><td>H9</td><td>M9</td><td>H4</td><td>M8</td><td>X9</td><td>T8</td><td>T3</td><td>H5</td><td>M1</td></tr>
+                    <tr><td>5</td><td>M1</td><td>M2</td><td>X1</td><td>M9</td><td>T9</td><td>M4</td><td>H3</td><td>H8</td><td>T4</td><td>M3</td></tr>
+                    <tr><td>6</td><td>H2</td><td>M1</td><td>T4</td><td>T7</td><td>X3</td><td>H6</td><td>M7</td><td>T5</td><td>H7</td><td>H4</td></tr>
+                    <tr><td>7</td><td>T8</td><td>H3</td><td>M2</td><td>X9</td><td>T2</td><td>H1</td><td>H4</td><td>M6</td><td>M1</td><td>T1</td></tr>
+                    <tr><td>8</td><td>M9</td><td>X2</td><td>H8</td><td>T8</td><td>M4</td><td>T3</td><td>M6</td><td>H7</td><td>H3</td><td>T8</td></tr>
+                    <tr><td>9</td><td>T7</td><td>T4</td><td>M4</td><td>H6</td><td>H7</td><td>M2</td><td>M5</td><td>T9</td><td>X7</td><td>H9</td></tr>
+                </table>
+                <ul>
+                    <li>T#: Tap the button when the last digit of the timer is a #</li>
```
Comment (Gameplushy) [inline Manual/HTML/Metered Button.html:121]: bomb timer?

File: Assets/ModScripts/Submodules/MeteredButton.cs [hunk 2]
```diff
@@ -0,0 +1,329 @@
...
+        if (Module.IsAnimating() || animating)
+            return;
+
+        Module.Audio.PlayGameSoundAtTransform(KMSoundOverride.SoundEffect.ButtonPress, Module.transform);
+        Module.UtilityButton.GetComponentInChildren<KMSelectable>().AddInteractionPunch(0.5f);
+
+        if (Module.IsModuleSolved())
+            return;
+
+        if (!Module.CheckValidComponents())
```
Comment (Gameplushy) [inline Assets/ModScripts/Submodules/MeteredButton.cs:170]: This is check every time big diamond is pressed. Maybe a getter for the Solving variable could be created?

**Review by Gameplushy — COMMENTED** [review]
Just a few changes to optimize/prettify some stuff. Might need to contact Sameone for an ambiguity in the manual

**Comment by Gameplushy** [comment]
At the current moment : The submodule cannot be generated on its own.
I have potentially identified a weakness (holding during a mashing sequence), but that's only speculation as I have not tested it yet. Good job however!

**Comment by Gameplushy** [comment]
At its current state, I believe it's good to go! Might want to wait for Sameone to get his opinion about the X8*9 rule before merging, just in case.

**Comment by GoodHood1** [comment]
> At its current state, I believe it's good to go! Might want to wait for Sameone to get his opinion about the X8*9 rule before merging, just in case.

**Comment by GoodHood1** [comment]
Sorry I managed to mislick on the close button

**Review by Gameplushy — APPROVED** [review]
Great! Good to merge

**Pull request merged**